middleware/compress_brotli.go

```diff
@@ -0,0 +1,121 @@
+package middleware
+
+import (
+	"bufio"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/andybalholm/brotli"
+
+	"github.com/labstack/echo/v4"
+)
+
+type (
+	// BrotliConfig defines the config for Brotli middleware.
+	BrotliConfig struct {
+		// Skipper defines a function to skip middleware.
+		Skipper Skipper
+
+		// Brotli compression level.
+		// Optional. Default value -1.
+		Level int `yaml:"level"`
+	}
+
+	brotliResponseWriter struct {
+		io.Writer
+		http.ResponseWriter
+	}
+)
+
+const (
+	brotliScheme = "br"
+)
+
+var (
+	// DefaultBrotliConfig is the default Brotli middleware config.
+	DefaultBrotliConfig = BrotliConfig{
+		Skipper: DefaultSkipper,
+		Level:   brotli.DefaultCompression,
+	}
+)
+
+// Brotli returns a middleware which compresses HTTP response using brotli compression
+// scheme.
+func Brotli() echo.MiddlewareFunc {
+	return BrotliWithConfig(DefaultBrotliConfig)
+}
+
+// BrotliWithConfig return Brotli middleware with config.
+// See: `Brotli()`.
+func BrotliWithConfig(config BrotliConfig) echo.MiddlewareFunc {
+	// Defaults
+	if config.Skipper == nil {
+		config.Skipper = DefaultBrotliConfig.Skipper
+	}
+	if config.Level == 0 {
```

Comment:
Be aware that for the Brotli library that you added a Level equal to 0 is a valid level. I thinkit is better to remove this check and keep the Level received by parameter.

https://github.com/andybalholm/brotli/blob/729edfbcfeaad43a6412c185205501ec80773250/writer.go#L8-L12
```go
const (
	BestSpeed          = 0
	BestCompression    = 11
	DefaultCompression = 6
)
```